### PR TITLE
[16.0] [MIG] Migrated partner_credit_limit

### PR DESCRIPTION
**Pull request by dhaval-serpentcs**

### CODE REVIEW

File: partner_credit_limit/LICENSE/LICENSE [hunk 1]
```diff
@@ -0,0 +1,17 @@
+    OpenERP, Open Source Management Solution
+    Copyright (C) 2018-Today Serpent Consulting Services Pvt. Ltd.
```
Comment (Trivedi-Vacha-SerpentCS) [inline partner_credit_limit/LICENSE/LICENSE]:
```suggestion
    Copyright (C) 2023-Today Serpent Consulting Services Pvt. Ltd.
```

File: partner_credit_limit/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# See LICENSE file for full copyright and licensing details.
+
+{
+    'name': 'Partner Credit Limit',
+    'version': '16.0.1.0.0',
+    'category': 'Partner',
+    'license': 'AGPL-3',
```
Comment (Trivedi-Vacha-SerpentCS) [inline partner_credit_limit/__manifest__.py]: [License](https://www.odoo.com/documentation/16.0/legal/licenses.html#odoo-16-community-edition)

File: partner_credit_limit/models/sale.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    _inherit = "sale.order"
+
+    def check_limit(self):
+        self.ensure_one()
+        partner = self.partner_id
+        user_id = self.env['res.users'].search([
+            ('partner_id', '=', partner.id)], limit=1)
+        if user_id and not user_id.has_group('base.group_portal') or not \
+                user_id:
+            moveline_obj = self.env['account.move.line']
```
Comment (Trivedi-Vacha-SerpentCS) [inline partner_credit_limit/models/sale.py:17]: define all self.env at the starting of the method

File: partner_credit_limit/models/sale.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+            movelines = moveline_obj.search(
```
Comment (Trivedi-Vacha-SerpentCS) [inline partner_credit_limit/models/sale.py:18]: use filter instead of search

File: partner_credit_limit/models/sale.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+                [('partner_id', '=', partner.id),
+                 ('account_id.account_type', 'in',
+                    ['Receivable', 'Payable']),
+                 ('parent_state','!=','cancel')]
+            )
+            confirm_sale_order = self.search(
```
Comment (Trivedi-Vacha-SerpentCS) [inline partner_credit_limit/models/sale.py:24]: use filter

File: partner_credit_limit/models/sale.py [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+                [('partner_id', '=', partner.id),
+                 ('state', '=', 'sale'),
+                 ('invoice_status', '!=', 'invoiced')])
+            debit, credit = 0.0, 0.0
+            amount_total = 0.0
+            for status in confirm_sale_order:
+                amount_total += status.amount_total
```
Comment (Trivedi-Vacha-SerpentCS) [inline partner_credit_limit/models/sale.py:31]: Use inline code:
amount_total = sum(sale.amount_total for sale in confirm_sale_order)

File: partner_credit_limit/models/sale.py [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+            for line in movelines:
+                credit += line.credit
+                debit += line.debit
```
Comment (Trivedi-Vacha-SerpentCS) [inline partner_credit_limit/models/sale.py:34]: same as above: use inline code

**Review by Trivedi-Vacha-SerpentCS — CHANGES_REQUESTED** [review]

File: partner_credit_limit/models/sale.py (hunk 6 above)
Comment (Vandan-Pandeji-SerpentCS) [reply partner_credit_limit/models/sale.py:31]: Use inline code:
amount_total = sum(confirm_sale_order.mapped('amount_total')))

File: partner_credit_limit/models/sale.py (hunk 7 above)
Comment (Vandan-Pandeji-SerpentCS) [reply partner_credit_limit/models/sale.py:34]: Use Inline Code:
credit = sum(movelines.mapped('credit'))

debit = sum(movelines.mapped('debit'))

**Comment by Vandan-Pandeji-SerpentCS** [comment]
@dhaval-serpentcs  please check the Review Feedback.

**Comment by dhaval-serpentcs** [comment]
i will improve feedback points and send pr from new branch.

**Pull request closed without merging**